.github/workflows/build-webapi.yaml

```diff
@@ -9,27 +9,49 @@ on:
     branches: [ dev, main ]
     paths:
       - 'src/dotnet/**'
+      - '.github/workflows/**'
 
 jobs:
-  build:
+  container-build-step:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ./src/dotnet/CarbonAware.WebApi
     steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+    
+    - name: Docker Target Build
+      run: docker build --target=build . -f CarbonAware.WebApi/Dockerfile -t ca-api-build
+      working-directory: ./src/dotnet
+
+    - name: Docker Unit Test
```

Comment:
After looking at the original yaml file, and running a couple quick local tests, I think we should go with this command, to run _"all the unit test" in one step_, also as a way to be forward-looking when other tests + test-projects get added to the repo : `docker run -w //src/ ca-api-build dotnet test`

Thoughts?

Reply:
I think that makes sense. I'll make the change.